Stripe/STPPaymentMethodsViewController.m

```diff
@@ -202,12 +217,21 @@ - (void)internalViewControllerDidDeletePaymentMethod:(id<STPPaymentMethod>)payme
     }
 }
 
-- (void)internalViewControllerDidCreateToken:(STPToken *)token completion:(STPErrorBlock)completion {
-    [self.apiAdapter attachSourceToCustomer:token completion:^(NSError *error) {
+- (void)internalViewControllerDidCreateSource:(id<STPSourceProtocol>)source completion:(STPErrorBlock)completion {
+    [self.apiAdapter attachSourceToCustomer:source completion:^(NSError *error) {
         stpDispatchToMainThreadIfNecessary(^{
             completion(error);
             if (!error) {
-                [self finishWithPaymentMethod:token.card];
+                // created a card token
+                if ([source isKindOfClass:[STPToken class]]) {
+                    STPToken *token = (STPToken *)source;
+                    [self finishWithPaymentMethod:token.card];
+                }
+                // created a card source
+                else if ([source isKindOfClass:[STPSource class]] &&
+                         ((STPSource *)source).type == STPSourceTypeCard) {
+                    [self finishWithPaymentMethod:(id<STPPaymentMethod>)source];
+                }
```

Comment:
I don't know the SDK well enough, is there a meaningful `else` clause that could be added? This, plus above in `finishWithPaymentMethod:`, make me a little nervous - but it looks okay to me.